edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -359,25 +361,28 @@ static string ToProxyLogLevel(LogEventLevel level)
 
         V1Volume GetVolume(KubernetesModule module, Mount mount)
         {
-            string volumeName = KubeUtils.SanitizeK8sValue(mount.Source);
+            // Volume name will be pvName if persistent volume is set, else will be volumeName
+            string volumeName = this.persistentVolumeName
+                .Map(KubeUtils.SanitizeK8sValue)
+                .GetOrElse(() => KubeUtils.SanitizeDNSValue(mount.Source));
+            // PVC name will be volume name
+            string pvcName = volumeName;
```

Comment:
In the past, we hit issue that two modules share the same mount name; and resulted in a clash. How are we addressing this scenario with this change?

Reply:
That will be a problem if the user chooses a storage class that only allows ReadWriteOnce.  If the user chooses to attempt the same volume twice in a deployment, and it's RWOnce, there's a conflict regardless of our implementation. 

I also discovered that part of the problem wasn't that K8s had a problem with 2 pods mounting the same PVC, the problem was that our code was trying to create the same PVC twice.  It was our fault all along. :(